### DISCUSSION TITLE
No such file or directory: 'ganache-cli' Lesson 5

### DISCUSSION THREAD

**Discussion by Bill-TheTroglodyte** [smartcontractkit/full-blockchain-solidity-course-py#1460] [Q&A]
Hello, at approximately 4:34:27 on Lesson 5, I have entered into my terminal on Visual Studio Code 'brownie run scripts/deploy.py' as prompted by the instructor. I get the following error after entering this information: [Errno 2] No such file or directory: 'ganache-cli'. I have properly installed ganache-cli and brownie and have triple checked multiple times. Please help me! Thank you.

I'm using:
mac OS
Python 3.10.4
node 16.15.0
Brownie v1.16.4
![Screen Shot 2022-05-03 at 2 19 05 PM](https://user-images.githubusercontent.com/100336752/167138768-e87faf66-b864-4202-8759-8854cc142dbc.png)
![Screen Shot 2022-05-03 at 2 58 45 PM](https://user-images.githubusercontent.com/100336752/167138807-47512318-f89c-4b99-be64-8f4fd1a5f732.png)

**Comment by Bill-TheTroglodyte** [comment]
I have read a bunch of material and tried the suggestions, but nothing is fixing it.

**Comment by cromewar** [comment]
Hello @Bill-TheTroglodyte 
Just some clarifying questions here:
1. Is you Mac a M1 chip one?
2. how did you installed ganache? if is installed on something like /usr/ this could cause troubles, nvm is the best option here.

**Reply by Bill-TheTroglodyte** [reply]
Thanks for replying!

I have a mid 2012 macbook pro.
Processor: 2.5 GHz Dual-Core Intel Core i5
I installed ganache using yarn on the command line.

**Reply by cromewar** [reply]
Ok based on your error I have to recomendations:
1. seems the private key for infura is not being recognized properly, so please use `export Infura_key=......` on you terminal before trying to run the code.
2. If the option above does not work, I think this could be most likely due to Python not being able to recognize your ganache installation, so I strongly suggest to reinstall nodejs trough nvm.

I hope any of this ingo might help.

**Reply by Bill-TheTroglodyte** [reply]
Thank you.

I tried both suggestions and neither worked. Is there a particular node version I should be running that works better with ganache?

**Reply by cromewar** [reply]
node version 16 lts, could you please just run ganache on your terminal and see if works?
and also, not likely to be the problem but anyways share for deploy code here.

**Reply by Bill-TheTroglodyte** [reply]
I tried to install ganache globally and got a bunch of errors(screenshot below). My terminal states that I have Ganache CLI v6.12.2 (ganache-core: 2.13.2) installed. I'm not sure which deploy code you're asking for. The code from VSCode in the screenshot above? Thank you for your help. I appreciate it very much
![Screen Shot 2022-05-06 at 10 42 24 AM](https://user-images.githubusercontent.com/100336752/167176296-22cd85a4-0da7-4467-a1c0-c81e25d9f581.png)

**Reply by cromewar** [reply]
@Bill-TheTroglodyte have you solved the problem?

**Reply by Bill-TheTroglodyte** [reply]
I have not. I tried your suggestions but I am running into the same issue.

**Reply by cromewar** [reply]
I most likely thing you have missing libraries on your macbook, have you installed Xcode and  cython?
And also does your mac have a m1 chip?

**Reply by Bill-TheTroglodyte** [reply]
My mac does not have an m1 chip. I installed both Xcode and cython.

**Reply by Bill-TheTroglodyte** [reply]
Cython version 0.29.28
